addons/sms/models/sms_sms.py

```diff
@@ -16,74 +17,72 @@ class SmsSms(models.Model):
     _order = 'id DESC'
 
     IAP_TO_SMS_STATE = {
+        'processing': 'process',
         'success': 'sent',
+        'delivered': 'delivered',  # not used for _postprocess_iap_results but webhook events.
         'insufficient_credit': 'sms_credit',
         'wrong_number_format': 'sms_number_format',
+        'country_not_supported': 'sms_country_not_supported',
         'server_error': 'sms_server',
         'unregistered': 'sms_acc'
     }
 
+    BOUNCE_DLR_ERRORS = {'sms_invalid_destination', 'sms_not_allowed', 'sms_rejected'}
+    DLR_ERRORS = {'sms_expired', 'sms_not_delivered', *BOUNCE_DLR_ERRORS}
+
+    uuid = fields.Char(
+        'UUID', help='UUID used by sms service', copy=False, readonly=True, index=True, default=lambda __: uuid4().hex,
+    )
     number = fields.Char('Number')
     body = fields.Text()
     partner_id = fields.Many2one('res.partner', 'Customer')
     mail_message_id = fields.Many2one('mail.message', index=True)
     state = fields.Selection([
         ('outgoing', 'In Queue'),
-        ('sent', 'Sent'),
+        ('process', 'Processing'),
+        ('sent', 'Sent'),  # Contrary to notifications and traces, this one does "only" mean sent.
+        ('delivered', 'Delivered'),
         ('error', 'Error'),
         ('canceled', 'Canceled')
     ], 'SMS Status', readonly=True, copy=False, default='outgoing', required=True)
     failure_type = fields.Selection([
+        ("unknown", "Unknown error"),
         ('sms_number_missing', 'Missing Number'),
         ('sms_number_format', 'Wrong Number Format'),
+        ('sms_country_not_supported', 'Country Not Supported'),
         ('sms_credit', 'Insufficient Credit'),
         ('sms_server', 'Server Error'),
         ('sms_acc', 'Unregistered Account'),
-        # mass mode specific codes
+        # mass mode specific codes, generated internally, not returned by IAP.
         ('sms_blacklist', 'Blacklisted'),
         ('sms_duplicate', 'Duplicate'),
         ('sms_optout', 'Opted Out'),
     ], copy=False)
+    sms_tracker_ids = fields.Many2one('sms.tracker', string='SMS trackers', compute='_compute_sms_tracker_ids')
+
+    _sql_constraints = [
+        ('uuid_unique', 'unique(uuid)', 'UUID must be unique'),
+    ]
+
+    @api.depends('uuid')
+    def _compute_sms_tracker_ids(self):
+        self.sms_tracker_ids = False
+        uuids = set(self.mapped('uuid'))
+        if not uuids:
+            return
+        existing_trackers = self.env['sms.tracker'].search([('sms_uuid', 'in', list(uuids))])
+        tracker_ids = {tracker.sms_uuid: tracker.id for tracker in existing_trackers}
+        for sms in self.filtered(lambda s: s.uuid in tracker_ids):
+            sms.sms_tracker_ids = self.env['sms.tracker'].browse(tracker_ids[sms.uuid])
```

Comment:
Don't really need to filter and to set to False above, you could just assign the ID or False with a .get